lesson21/src/js/main.js

```diff
@@ -0,0 +1,216 @@
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtonAttributes = [
+    {dataOrder:"desc",src:"../img/asc.svg", alt:"asc-image"},
+    {dataOrder:"both",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"asc",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && th.appendChild(createSortButtons(userKey));
+    }
+}
+
+const createSortButtons = (columnKey) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-order':button.dataOrder
+            }
+        });
+
+        button.dataOrder !== "asc" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassSortButton = (sortButtons,index) => {
+    sortButtons[index].classList.add("hidden");
+    (index === sortButtons.length -1) ? index = 0 : index++;
+    sortButtons[index].classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (key, usersData, dataOrder) => {
+    const cloneUsersData = [...usersData];
+
+    const sortButtonsFunc = {
+        "both": () => {
+            updateTableBody(usersData);
+        },
+        "asc" : () => {
+            cloneUsersData.sort((a,b) => a[key] - b[key]);
+            updateTableBody(cloneUsersData);
+        },
+        "desc": () => {
+            cloneUsersData.sort((a,b) => b[key] - a[key]);
+            updateTableBody(cloneUsersData);
+        }
+    }
+
+    Object.keys(sortButtonsFunc).forEach((button) => dataOrder === button && sortButtonsFunc[button]() );
+}
+
+const sortTableBody = usersData => {
+    for(const key of Object.keys(sortByUsersTableColumn)){
+        const sortButtons = [...document.getElementsByClassName(`js-${key}SortButton`)];
+        const sortButtonsBox = document.getElementById(`js-${key}Buttons-Box`);
+        
+        let dataOrder;
```

Comment:
Q.

Reply:
As you pointed out, I don't need to use let!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/45d7b7e80e46fb31181ba23ae22051b372cdba5e